pkg/webhook/validatingwebhook/validate.go

```diff
@@ -0,0 +1,144 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+	deserializer  = codecs.UniversalDeserializer()
+
+	log = logf.Log.WithName("users_rolebindings_validating_webhook")
+)
+
+type Validator struct {
+	Client runtimeClient.Client
+}
+
+func (v Validator) HandleValidate(w http.ResponseWriter, r *http.Request) {
+	var respBody []byte
+	body, err := ioutil.ReadAll(r.Body)
+	defer func() {
+		if err := r.Body.Close(); err != nil {
+			log.Error(err, "unable to close the body")
+		}
+	}()
+	if err != nil {
+		log.Error(err, "unable to read the body of the request")
+		w.WriteHeader(http.StatusInternalServerError)
+		respBody = []byte(fmt.Sprintf("unable to read the body of the request: %s", err))
+	} else {
+		// validate the request
+		respBody = validate(body, v.Client)
+		w.WriteHeader(http.StatusOK)
+	}
+	if _, err := w.Write(respBody); err != nil {
+		log.Error(err, "unable to write response")
+	}
+}
+
+func validate(body []byte, client runtimeClient.Client) []byte {
+	admReview := v1.AdmissionReview{}
+	if _, _, err := deserializer.Decode(body, nil, &admReview); err != nil {
+		log.Error(err, "unable to deserialize the admission review object", "body", body)
+		return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable to deserialize the admission review object - body: %v", body))
+	}
+	// let's unmarshal the object to be sure that it's a rolebinding
+	var rb *rbac.RoleBinding
+	if err := json.Unmarshal(admReview.Request.Object.Raw, &rb); err != nil {
+		log.Error(err, "unable unmarshal rolebinding json object", "AdmissionReview", admReview)
+		return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable unmarshal rolebinding json object - raw request object: %v", admReview.Request.Object.Raw))
+	}
+	requestingUsername := admReview.Request.UserInfo.Username
+	// allow admission request if the user is a system user
+	if strings.HasPrefix(requestingUsername, "system:") {
+		return allowAdmissionRequest(admReview)
+	}
+	subjects := rb.Subjects
+	allServiceAccountsSubject := rbac.Subject{
+		Kind:     "Group",
+		Name:     "system:serviceaccounts",
+		APIGroup: "rbac.authorization.k8s.io",
+	}
+	allUsersSubject := rbac.Subject{
+		Kind:     "Group",
+		Name:     "system:authenticated",
+		APIGroup: "rbac.authorization.k8s.io",
+	}
+	for _, sub := range subjects {
+		if sub == allUsersSubject || sub == allServiceAccountsSubject {
+			requestingUser := &userv1.User{}
+			err := client.Get(context.TODO(), types.NamespacedName{
+				Name: admReview.Request.UserInfo.Username,
+			}, requestingUser)
+
+			if err != nil {
+				log.Error(fmt.Errorf("Cannot find the user: %w", err), "unable to find the user requesting creation")
+				return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable to find the user requesting creation: %s", requestingUsername))
```

Comment:
As we agreed (multiple times :) We do not want to deny if we it's unknown user (for whatever reason). Check the error. If it's NotFound than log it as a warning/info and allow the request. If it's something else then we can deny with the error message.

Reply:
right, my apologies! It wasn't coming intuitively to me to allow after logging error, and i kept missing it. Thank you for pointing this out. fixed in 9d1180e

On a separate note, thanks to this error, i was able to find out that i need to create a service account for validating webhook to access kube client.